### PR TITLE
MHV-47337 Enable DataDogRUM for MHV Secure Messaging

### PR DESCRIPTION
**Pull request by alex-morgun-1**
## Summary

- incorporate DataDogRUM configurations in to Secure Messaging application, to enable UI Analytics tracking via DataDog
- Add privacy masks to all sensitive fields that may include any PII, PHI data. 

## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-47337

## What areas of the site does it impact?

My Health - Secure Messages

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx [hunk 1]
```diff
@@ -117,7 +117,7 @@ const ReplyForm = props => {
           subject,
         };
         if (draft && replyToMessageId) {
-          messageData[`${'draft_id'}`] = replyToMessageId; // if replying to a thread that has a saved draft, set a draft_id field in a request body
+          messageData[`${'draft_id'}`] = replyMessage.messageId; // if replying to a thread that has a saved draft, set a draft_id field in a request body
```
Comment (mdewey) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx:120]: QUESTION: Should this have gotten changed? It looks like `replyToMessageId` can be a [few different values](https://github.com/department-of-veterans-affairs/vets-website/blob/b8c00c8e2da55bb84fe7453daa09b4ffa873305f/src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx#L57).

File: src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx [hunk 2]
```diff
@@ -126,7 +126,7 @@ const ReplyForm = props => {
           sendData.append('message', JSON.stringify(messageData));
           attachments.map(upload => sendData.append('uploads[]', upload));
 
-          dispatch(sendReply(replyToMessageId, sendData, true))
+          dispatch(sendReply(replyMessage.messageId, sendData, true))
```
Comment (mdewey) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx:129]: DUPLICATE: same as about around `replyToMessageId`

**Review by mdewey — CHANGES_REQUESTED** [review]
The DDRUM looks good to me, a same question about a variable change

**Review by mdewey — APPROVED** [review]
LGTM

**Review by vsaleem — APPROVED** [review]

**Pull request merged**